src/app/shared/services/openWeatherMaps.js

```diff
@@ -0,0 +1,60 @@
+export default class OpenWeatherMaps {
+
+    constructor($http, fiveDayForecast, latestError, currentForecast) {
+        this.$http = $http;
+        this.fiveDayForecast = fiveDayForecast;
+        this.latestError = latestError;
+        this.currentForecast = currentForecast;
+        this.baseURL = 'https://api.openweathermap.org/data/2.5';
+    }
+
+    async getCurrentForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/weather';
+
+        const params = {
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        return this.$http.get(URL, params)
+            .then((response) => {
+                this.currentForecast = response;
+            })
+            .catch((response) => {
+                this.latestError = response;
+            });
+    };
+
+    async getFiveDayForecastCity(city) {
+        const params = {
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        try {
+            return await this.$http.get(`${this.baseURL}/forecast`, params).then((response) => {
```

Comment:
what? Don't mix await with resolving the promise via then.

Reply:
Havent finished yet, sorry for commiting!